contract/contracts/PixelDevsUkraineDonation.sol

```diff
@@ -80,11 +83,16 @@ contract PixelDevsUkraineDonation is
     }
 
     function withdraw() public onlyOwner {
-        payable(msg.sender).transfer(address(this).balance);
+        payable(withdrawWallet).transfer(address(this).balance);
```

Comment:
This makes me wonder if we should gate this by checking if payee is the withdrawer, rather than as `onlyOwner`, so we're not the middleperson. @rasmuscnielsen @goldzulu what do you think?

Not related to your changes, since the existing `withdraw()` used the `payable().transfer()` pattern, but I've heard mixed news about using this method. Maybe @goldzulu you can comment on this? For example, here's suggestions from https://solidity-by-example.org/sending-ether/

Reply:
I haven't read through your link yet but my initial idea was that ANYONE should actually be able to invoke the withdraw method. This way we're indeed not the middle person and anyone can put it on themselves to trigger the donation to happen.

Are there any downsides to this that I'm missing?

Reply:
Yes the call method probably is best nowadays however need to make sure reentrancy is guarded. On withdrawal need to be careful we are not opening up to anyone to be able withdraw to their own address. I feel safer with @carlomigueldy having quick look at this.

Reply:
Let me know your thoughts. If the transfer function limits gas to 2300, shouldn't reentrancy not be possible? I can always at a mutex if needed.

Reply:
So @rasmuscnielsen do you mean just removing `onlyOwner` guard? That would then let anyone transfer the funds to UkraineDAO.

I was initially thinking that `onlyOwner` would be replaced with a new guard that would confirm UkraineDAO was calling it, but maybe that's overkill and if `withdraw()` can only send funds to one place, removing `onlyOwner` seems fine.

Reply:
Hey guys @goldzulu @itstargetconfirmed @briangershon @rasmuscnielsen , we can use `AccessControl` from OpenZeppelin. That way we can add anyone that should have access to calling `withdraw`.

https://docs.openzeppelin.com/contracts/4.x/access-control

We can replace `Ownable`  with `AccessControl` implementation and add multiple wallet addresses, so it not only limits to `onlyOwner` that can call `withdraw`

With `AccessControl` we will use the modifier `onlyRole(DEFAULT_ADMIN_ROLE)` so for all wallet address that we add or grant the role of admin, only them with suffice permissions can call `withdraw`

Here be sample snippet
```solidity
contract PixelDevsUkraineDAO is AccessControl {
  constructor() {
    _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
    _grantRole(DEFAULT_ADMIN_ROLE, 0xgoldzulu);
    _grantRole(DEFAULT_ADMIN_ROLE, 0xbrian);
    _grantRole(DEFAULT_ADMIN_ROLE, 0xram);
    _grantRole(DEFAULT_ADMIN_ROLE, 0xitstarget);
  }

  function withdraw() external onlyRole(DEFAULT_ADMIN_ROLE) {
    // withdrawal logic
  }
} 
```

Reply:
Thanks @carlomigueldy we probably also need a safe way to allow anyone to trigger withdrawal. Though thinking about it now I think it might be a good idea. Withdrawal address need to be mutable with a default set initially on deployment. Can you suggest modifiers to and controls set in place so only contract owner can set and NOONE else please?

Reply:
For the withdraw() method I think it might be overkill to introduce roles. At least I can't think of a scenario where we want to limit who can call it? 
More than anything, opening it up would allow us all to feel rest assured that anyone can put it on themselves to make sure the donations are in fact sent off.

Reply:
> Thanks @carlomigueldy we probably also need a safe way to allow anyone to trigger withdrawal. Though thinking about it now I think it might be a good idea. Withdrawal address need to be mutable with a default set initially on deployment. Can you suggest modifiers to and controls set in place so only contract owner can set and NOONE else please?

Yep, the modifier `onlyRole` which comes from `AccessControl` will handle that. So it only limits to us that can mutate the `withdrawalAddress` 

Here be an example
```solidity
function withdraw() external onlyRole(DEFAULT_ADMIN_ROLE) {}
```

Reply:
> For the withdraw() method I think it might be overkill to introduce roles. At least I can't think of a scenario where we want to limit who can call it? More than anything, opening it up would allow us all to feel rest assured that anyone can put it on themselves to make sure the donations are in fact sent off.

Yeah if using `AccessControl` would be over kill then the `Ownable` contract to inherit should be enough if we only allow 1 wallet address to mutate the `withdrawalAddress`, because there can be a chance one of the wallet address that have `DEFAULT_ADMIN_ROLE` role attached they can possibly steal the funds. 

But not saying someone would actually do that, but there's a possibility. Though we can still identify who because transactions are transparent.

Reply:
> Thanks @carlomigueldy we probably also need a safe way to allow anyone to trigger withdrawal. Though thinking about it now I think it might be a good idea. Withdrawal address need to be mutable with a default set initially on deployment. Can you suggest modifiers to and controls set in place so only contract owner can set and NOONE else please?

Yep, the modifier `onlyRole` which comes from `AccessControl` will handle that. 

We can add roles too and they what they can only do is call `withdraw` then for setting the `withdrawalAddress` then we can only allow that to owner/deployer or a wallet address with `DEFAULT_ADMIN_ROLE` attached.

Here be an example

```solidity
// create new role specifically for calling withdraw
bytes32 public constant WITHDRAW_ROLE = keccak256("WITHDRAW_ROLE");

constructor() {
  // who can call setWithdrawalAddress method
  _grantRole(DEFAULT_ADMIN_ROLE, 0xgoldzulu); 

  // who can call withdraw method
  _grantRole(WITHDRAW_ROLE, 0xgoldzulu);
  _grantRole(WITHDRAW_ROLE, 0xcaromigueldy);
  _grantRole(WITHDRAW_ROLE, 0xbrianfive);
  _grantRole(WITHDRAW_ROLE, 0xrasmuscnielsen);
  _grantRole(WITHDRAW_ROLE, 0xitstargetconfirmed);
}

function withdraw(address _withdrawalAddress) external {
  require(
    hasRole(WITHDRAW_ROLE, msg.sender), 
    "You don't have permissions to withdraw."
  );

  // withdraw logic
}

function setWithdrawalAddress(address _withdrawalAddress) 
  external 
  onlyRole(DEFAULT_ADMIN_ROLE) 
{
  withdrawalAddress = _withdrawalAddress;
}
```

cc @goldzulu @rasmuscnielsen @briangershon @itstargetconfirmed

Reply:
Furthermore, I don't think using `AccessControl` will be an overkill but is necessary to what we are trying to achieve